### PR TITLE
fix collectXhtml regex

### PR DESCRIPTION
**Pull request by fansanelli**
| Bugfix        | yes

### Description

create a feed with laminas-feed and try to consume it.

### CODE REVIEW

File: src/Reader/Extension/Atom/Entry.php [hunk 1]
```diff
@@ -142,8 +142,8 @@ protected function collectXhtml($xhtml, $prefix)
             $prefix = $prefix . ':';
         }
         $matches = [
-            '/<\?xml[^<]*>[^<]*<' . $prefix . 'div[^<]*/',
-            '/<\/' . $prefix . 'div>\s*$/',
+            '/<\?xml[^<]*>[^<]*<' . $prefix . 'div[^>]*>/',
+            '/<\/' . $prefix . 'div>\s*$/'
```
Comment (michalbundyra) [inline src/Reader/Extension/Atom/Entry.php]: Please add back the trailing comma.

**Comment by fansanelli** [comment]
> @fansanelli Thanks for your contribution.
> 
> Can you provide a failing test case - the best if you just add unit test to cover your change. Thanks.
```
<?xml version="1.0" encoding="UTF-8"?>
<xhtml:div xmlns:xhtml="http://www.w3.org/1999/xhtml">XXX</xhtml:div>
```
should become
```
XXX
```

right?

**Comment by fansanelli** [comment]
i created units but TBH haven't actually run them... can you please review them?

**Comment by fansanelli** [comment]
Travis is now happy.. Now you can do a proper review

File: test/Reader/Entry/AtomTest.php [hunk 2]
```diff
@@ -360,6 +360,15 @@ public function testGetsContentFromAtom10XhtmlNamespaced()
         $this->assertEquals('<p class="x:"><em>Entry Content &amp;x:</em></p>', $entry->getContent());
     }
 
+    public function testGetsContentFromAtom10XhtmlNamespacedAlt()
```
[the remark this answers is not in the document]
Comment (fansanelli) [reply test/Reader/Entry/AtomTest.php]: testGetsContentFromAtom10XhtmlNamespacedTextOnly?

File: test/Reader/Entry/AtomTest.php [hunk 3]
```diff
@@ -360,6 +360,15 @@ public function testGetsContentFromAtom10XhtmlNamespaced()
...
+    {
+        $feed  = Reader\Reader::importString(
+            file_get_contents($this->feedSamplePath . '/content/plain/atom10_Xhtml_alt.xml')
```
[the remark this answers is not in the document]
Comment (fansanelli) [reply test/Reader/Entry/AtomTest.php]: atom10_Xhtml_text_only.xml?

File: test/Reader/Entry/AtomTest.php [hunk 4]
```diff
@@ -360,6 +360,15 @@ public function testGetsContentFromAtom10XhtmlNamespaced()
...
+        );
+        $entry = $feed->current();
+        $this->assertEquals('XXX', $entry->getContent());
```
[the remark this answers is not in the document]
Comment (fansanelli) [reply test/Reader/Entry/AtomTest.php]: s/XXX/Entry Content/g ?

File: test/Reader/Entry/AtomTest.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (froschdesign) [reply test/Reader/Entry/AtomTest.php]: Suggestion: `testGetsContentWithoutChildElementsFromAtom10XhtmlNamespaced`

File: test/Reader/Entry/_files/Atom/content/plain/atom10_Xhtml_alt.xml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<feed xmlns="http://www.w3.org/2005/Atom">
+    <entry>
+		<content xmlns:xhtml="http://www.w3.org/1999/xhtml" type="xhtml">
+			<xhtml:div xmlns:xhtml="http://www.w3.org/1999/xhtml">XXX</xhtml:div>
+		</content>
+    </entry>
+</feed>
```
Comment (froschdesign) [inline test/Reader/Entry/_files/Atom/content/plain/atom10_Xhtml_alt.xml]: For consistency with other unit test I would suggest the following file content:

```xml
<?xml version="1.0" encoding="utf-8"?>
<feed xmlns="http://www.w3.org/2005/Atom">
    <entry>
        <content type="xhtml" xmlns:x="http://www.w3.org/1999/xhtml">
            <x:div>Entry Content &amp;x:</x:div>
        </content>
    </entry>
</feed>
```

**Comment by fansanelli** [comment]
let's see if Travis is still happy

**Comment by fansanelli** [comment]
Any other comments or blockers?

**Comment by fansanelli** [comment]
Humble ping. Can this be merged?

**Comment by froschdesign** [comment]
@fansanelli 
> Any other comments or blockers?

Yes, the update of the XML file is still missing. See my comment above: https://github.com/laminas/laminas-feed/pull/17#discussion_r390318599

> Humble ping. Can this be merged?

It is not necessary to point to open tasks. The tools here on GitHub do that all by themselves. Your pull request is therefore not forgotten! 😃

**Comment by fansanelli** [comment]
No.. I updated it...

**Comment by froschdesign** [comment]
@fansanelli 
Sorry, I mean you should use the entire content of my suggestion for consistency with the content of:

https://github.com/laminas/laminas-feed/blob/master/test/Reader/Entry/_files/Atom/content/plain/atom10_Xhtml.xml

**Comment by fansanelli** [comment]
My example is generated by laminas itself... That's why I did it that way

**Comment by froschdesign** [comment]
@fansanelli 
> My example is generated by laminas itself...

I know this because I tested it myself. But not only the self-produced content of this library should be tested.
Anyway you don't have to change it, I can change it on merging.

**Comment by fansanelli** [comment]
@froschdesign if you can change it on merging, you can proceed... if you need me touch something let me know.

**Comment by michalbundyra** [comment]
Thanks, @fansanelli!

**Pull request merged**